### PR TITLE
feat: more tooling

### PR DESCRIPTION
**Pull request by MarioSimou**

### CODE REVIEW

File: .github/workflows/codequality.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Code Quality
```
Comment (MarioSimou) [inline .github/workflows/codequality.yml:1]: I put those steps into a reusable workflow because I am expecting to reuse it in the master pipeline, before the deployment

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -34,29 +22,21 @@ jobs:
...
-          PORT: '8080'
-          SERVER_URL: http://localhost:8080
-          AUTH0_CLIENT_ID: ${{ secrets.AUTH0_CLIENT_ID }}
-          AUTH0_CLIENT_SECRET: ${{ secrets.AUTH0_CLIENT_SECRET }}
-          SESSION_SECRET: super-duper-s3cret
-          AUTH0_DOMAIN: nevados-dev.us.auth0.com
-          AUTH0_ORGANIZATION: org_3pmM22dWDm2Ppb95
-          AUTH0_TEST_USERNAME: test@nevados.solar
-          AUTH0_TEST_PASSWORD: test22!
+          GH_APP_ID: ${{ secrets.GH_APP_ID }}
```
Comment (MarioSimou) [inline .github/workflows/pull-request.yml:35]: I am not sure if I am referencing the right secrets names because I don't have access to them. When they change, I expect the pipeline to pass

File: package.json [hunk 3]
```diff
@@ -17,12 +17,20 @@
     "build:remix": "remix build",
     "dev": "run-p dev:*",
     "dev:css": "npm run generate:css -- --dev",
-    "dev:remix": "remix dev",
+    "dev:remix": "PORT=8080 remix dev",
```
Comment (MarioSimou) [inline package.json:20]: Adjust port to 8080 so its aligned with what we are running in e2e and avoid confusion

Comment (PatrickHeneise) [reply package.json:20]: `PORT` should always be set in the environment, not in a script/command.

File: .github/workflows/codequality.yml (hunk 1 above)
Comment (PatrickHeneise) [reply .github/workflows/codequality.yml:1]: Interesting approach; not sure what to think of it though. The re-usable workflow runs a lot of duplicate commands

File: .npmrc [hunk 4]
```diff
@@ -1 +1,2 @@
-legacy-peer-deps=true
+strict-peer-dependencies=false
+auto-install-peers=false
```
Comment (PatrickHeneise) [inline .npmrc:2]: Please add newline at the end

File: .prettierignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+build/
+node_modules/
+app/styles/windi.css
+.pnpm-store
```
Comment (PatrickHeneise) [inline .prettierignore:4]: Please add newline at the end

File: .prettierrc [hunk 6]
```diff
@@ -1,6 +1,9 @@
 {
   "singleQuote": true,
   "semi": false,
-  "trailingComma": "none",
-  "proseWrap": "always"
+  "trailingComma": "es5",
+  "proseWrap": "always",
+  "bracketSpacing": true,
```
Comment (PatrickHeneise) [inline .prettierrc:6]: I'm coming from standard-js, so the original settings are most compatible between the two. Would like to keep the opinionated settings to an absolute minimum.

File: app/components/Groups.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import { render, screen } from '@testing-library/react'
```
Comment (PatrickHeneise) [inline app/components/Groups.spec.tsx:1]: most examples in https://github.com/vitest-dev/vitest/tree/main/examples use a dedicated `test` folder, I'd prefer that pattern over mixing code and tests

File: cypress/cypress.config.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { defineConfig } from 'cypress'
```
Comment (PatrickHeneise) [inline cypress/cypress.config.ts:1]: Cypress is pretty heavy weight, maybe Playwright is a better/lighter option for this project to test the basics?

File: package.json [hunk 9]
```diff
@@ -17,12 +17,20 @@
...
     "generate:css": "npx windicss './app/**/*.{jsx,tsx}' -to ./app/styles/windi.css -f windi.config.js",
-    "start": "remix-serve build",
-    "lint": "eslint --cache --cache-location ./node_modules/.cache/eslint .",
+    "start": "PORT=8080 remix-serve build",
```
Comment (PatrickHeneise) [inline package.json:22]: `PORT` is an environment variable. Not all runners use 8080 and this can break deployment.

File: package.json (hunk 3 above)
Comment (MarioSimou) [reply package.json:20]: I will remove it, but is there a specific reason why we had to use port 8080 in e2e tests? We could use 3000 and avoid having the discrepancy running remix on a different port for diffferent scenarios.

File: .npmrc (hunk 4 above)
Comment (MarioSimou) [reply .npmrc:2]: I followed the official pnpm documentation [here](https://pnpm.io/npmrc#peer-dependency-settings). My understanding says that `strict-peer-dependencies` and  `auto-install-peers` are the replacement of `legacy-peers-deps` flag

File: .prettierrc (hunk 6 above)
Comment (MarioSimou) [reply .prettierrc:6]: Okay. I will revert that

File: app/components/Groups.spec.tsx (hunk 7 above)
Comment (MarioSimou) [reply app/components/Groups.spec.tsx:1]: Okay, but what's the indentend file structure we want to follow? 

I have seen this in the past: 
```
[app]
   [routes]
        index.tsx
        
[test]
   [app]
      [routes]
          index.spec.tsx
```

At the moment I use this, which is the reason I thought the test directory is redundant:
```
[app]
   [routes]
        [home]
            index.tsx
            index.spec.tsx
```

File: cypress/cypress.config.ts (hunk 8 above)
Comment (MarioSimou) [reply cypress/cypress.config.ts:1]: I haven't used it in the past, but I can give it a try

File: package.json (hunk 9 above)
Comment (MarioSimou) [reply package.json:22]: I will remove the port here

File: .npmrc (hunk 4 above)
Comment (PatrickHeneise) [reply .npmrc:2]: Sorry, the comment is related to the newline character at the end of the file. GitHub shows a big red symbol in the code preview then the newline character is missing, which is ugly

Screenshot 2022-09-07 at 12 36 05 PM

File: package.json (hunk 3 above)
Comment (PatrickHeneise) [reply package.json:20]: The tests are removed; that was a copy&paste error.

File: .github/workflows/codequality.yml (hunk 1 above)
Comment (MarioSimou) [reply .github/workflows/codequality.yml:1]: I also tend to have two pipelines, but I run the codequality, unit and e2e stages again in master. It sounds a bit redundant but it gives me more security. For example, you won't test anything of the checks we do when you push straight to main branch. I use CI to give me that extra layer of security. I also like the workflow diagrams because they are very easy to undestand.  

Screenshot 2022-09-07 at 10 46 19

In summary, the pipeline can be a bit slower to deploy in prod, but for a small project like this I don't expect to make a noticable difference.

File: cypress/cypress.config.ts (hunk 8 above)
Comment (MarioSimou) [reply cypress/cypress.config.ts:1]: @PatrickHeneise I will remove the cypress setup on this pr and raise a following one usng `Playwright` or `Cypress`. It will depend on the spike output

File: .github/workflows/codequality.yml (hunk 1 above)
Comment (PatrickHeneise) [reply .github/workflows/codequality.yml:1]: We can try it out, see how it works.

File: app/components/Groups.spec.tsx (hunk 7 above)
Comment (PatrickHeneise) [reply app/components/Groups.spec.tsx:1]: test-app-routes or test-routes should work

**Review by PatrickHeneise — APPROVED** [review]

**Comment by PatrickHeneise** [comment]
Thanks @MarioSimou !
